src/drivers/ports/esp32/can_port.i

```diff
@@ -39,6 +39,82 @@
 static struct fs_counter_t rx_channel_overflow;
 static struct fs_counter_t errors;
 
+extern xSemaphoreHandle thrd_idle_sem;
+
+/**
+ * Check if module is waiting for bus idle.
+ *
+ * Note 3 to Status register description in SJA1000 reference:
+ *     "If both the receive status and the transmit status bits are
+ *      logic 0 (idle) the CAN-bus is idle. If both bits are set the
+ *      controller is waiting to become idle again..."
+ */
+static int is_waiting_for_bus_idle(uint32_t status_reg_value)
+{
+    return ((status_reg_value & (ESP32_CAN_STATUS_RX | ESP32_CAN_STATUS_TX)) ==
+                                (ESP32_CAN_STATUS_RX | ESP32_CAN_STATUS_TX));
+}
+
+/**
+ *  Wait some time for bus is idle.
+ *
+ * @param[in] self_p Driver object to work with.
+ *
+ * @return zero(0) if success or (-ENETDOWN) is case of timeout.
+ *
+ * CAN-module can be in state waiting for bus idle. At reset it waits for
+ * 11 consecutive recessive bits, so we will wait for at least 170 bit-periods
+ * (128*1.2+11 = 164, where 128 - is maximum CAN-frame length, 1.2 - bit stuffing koeff).
+ * In transition from bus-off to bus-on state it waits by default for 128 periods of
+ * 11 consecutive recessive bits but this number 128 is readed from TXERR register. So
+ * this function will work in all situations if TXERR is cleared.
+ */
+static int wait_for_bus_idle(volatile struct esp32_can_t *regs_p)
+{
+    struct time_t now, stop, delta;
+
+    time_get(&now);
+    delta.seconds = 0;
+    delta.nanoseconds = 17000L * 1000;    // 17000 = 170 * 100: 170 cycles of bit-period @ 10 KBPS (100 us per bit)
+    time_add(&stop, &now, &delta);
+    while ((now.seconds < stop.seconds) ||
```

Comment:
Add a blank line before while statement. Always have a blank line before and after statements.